### PR TITLE
feat: store release notes in mongodb

### PR DESCRIPTION
**Pull request by Skisocks**
This PR adds a MongoDB instance to the service's charts. No actual functionality has been added - the service just connects to the instance and pings it.

### CODE REVIEW

File: charts/peacock/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,11 @@
-apiVersion: v1
+apiVersion: v2
 description: A Helm chart for Peacock Service on Kubernetes
 icon: https://raw.githubusercontent.com/spring-financial-group/peacock/main/images/peacock.png
 name: peacock
 version: 0.1.0-SNAPSHOT
+
+dependencies:
+  - name: mongodb
+    version: '^13' # Any 13.x.x version, but not version 12 or 11
```
Comment (tomhobson) [inline charts/peacock/Chart.yaml]: do we need this?

File: charts/peacock/values.yaml [hunk 2]
```diff
@@ -17,9 +17,37 @@ webhookSecret: ""
 # Token for authenticating the message webhook
 webhookToken: ""
 
+# The connection string used for the database
+mongodbConnectionString: ""
```
Comment (tomhobson) [inline charts/peacock/values.yaml:21]: Let's describe that this is an override + How to use it?

Worth putting a chart readme?

**Review by tomhobson — CHANGES_REQUESTED** [review]

File: pkg/release/delivery/releasehandler.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	releases, err := h.releaseUc.GetReleasesAfterDate(c, environment, startTime)
+	if err != nil {
+		c.JSON(500, gin.H{"error": err.Error()})
+		return
+	}
+
+	c.JSON(200, models.GetReleasesResponse{
+		Releases: releases,
```
Comment (tomhobson) [inline pkg/release/delivery/releasehandler.go:48]: Hmm should we curate the release notes or are we going to filter them on the client side?

ie curated for different user types?

Should we add a filter for the type of release notes you want to see?

File: pkg/release/usecase/releaseuc.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	repository domain.ReleaseRepository
+}
+
+func NewUseCase(repository domain.ReleaseRepository) domain.ReleaseUseCase {
+	return &useCase{
+		repository: repository,
+	}
+}
+
+func (uc *useCase) SaveRelease(ctx context.Context, environment string, releaseNotes []models.ReleaseNote) error {
```
Comment (tomhobson) [inline pkg/release/usecase/releaseuc.go:20]: Should we tie it back to a PR?

**Review by tomhobson — CHANGES_REQUESTED** [review]

File: pkg/release/usecase/releaseuc.go (hunk 4 above)
Comment (ZackJagger) [reply pkg/release/usecase/releaseuc.go:20]: yeah, sounds gooood

File: pkg/release/delivery/releasehandler.go (hunk 3 above)
Comment (ZackJagger) [reply pkg/release/delivery/releasehandler.go:48]: Yeah, just to clarify, by type, do you mean the Teams referenced in the release notes? @tomhobson

Comment (tomhobson) [reply pkg/release/delivery/releasehandler.go:48]: yes sorry, filter by feathers i mean. not using the peacock lingo

File: pkg/release/usecase/releaseuc.go (hunk 4 above)
Comment (ZackJagger) [reply pkg/release/usecase/releaseuc.go]: sorted

File: pkg/release/delivery/releasehandler.go (hunk 3 above)
Comment (ZackJagger) [reply pkg/release/delivery/releasehandler.go:58]: implemented!

**Review by tomhobson — APPROVED** [review]

**Pull request merged**